sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -266,7 +266,12 @@ abstract class CastBase extends UnaryExpression with TimeZoneAwareExpression wit
       TypeCheckResult.TypeCheckSuccess
     } else {
       TypeCheckResult.TypeCheckFailure(
-        s"cannot cast ${child.dataType.catalogString} to ${dataType.catalogString}")
+        if ( child.dataType.isInstanceOf[NumericType] && dataType.isInstanceOf[TimestampType]) {
```

Comment:
Please, remove spaces:
```scala
if (child.dataType.isInstanceOf[NumericType] && dataType.isInstanceOf[TimestampType])
```
Take a look at the style guide: https://github.com/databricks/scala-style-guide

Reply:
yes,i will correct it.